data/pigui/modules/info-view/01-ship-info.lua

```diff
@@ -0,0 +1,127 @@
+-- Copyright © 2008-2018 Pioneer Developers. See AUTHORS.txt for details
+-- Licensed under the terms of the GPL v3. See licenses/GPL-3.txt
+
+local ui = import 'pigui/pigui.lua'
+local InfoView = import 'pigui/views/info-view'
+local Lang = import 'Lang'
+local Game = import 'Game'
+local Vector = import 'Vector'
+local ShipDef = import 'ShipDef'
+local Equipment = import 'Equipment'
+
+local pionillium = ui.fonts.pionillium
+local orbiteer = ui.fonts.orbiteer
+
+local drawTable = import 'pigui/libs/table.lua'
+
+local l = Lang.GetResource("ui-core")
+
+local function drawShipEquipment()
+	ui.withFont(orbiteer.large.name, orbiteer.large.size, function()
+		ui.text(l.EQUIPMENT)
+	end)
+	ui.text("")
+	local equips = {Equipment.cargo, Equipment.misc, Equipment.hyperspace, Equipment.laser}
+	for _,t in pairs(equips) do
+		for k,et in pairs(t) do
+			local slot = et:GetDefaultSlot(Game.player)
+			if (slot ~= "cargo" and slot ~= "missile" and slot ~= "engine" and slot ~= "laser_front" and slot ~= "laser_rear") then
+				local count = Game.player:CountEquip(et)
+				if count == 1 then
+					ui.text(et:GetName())
+				elseif count > 1 then
+					if et == Equipment.misc.shield_generator then
+						ui.text(string.interp(l.N_SHIELD_GENERATORS, { quantity = string.format("%d", count) }))
+					elseif et == Equipment.misc.cabin_occupied then
+						ui.text(string.interp(l.N_OCCUPIED_PASSENGER_CABINS, { quantity = string.format("%d", count) }))
+					elseif et == Equipment.misc.cabin then
+						ui.text(string.interp(l.N_UNOCCUPIED_PASSENGER_CABINS, { quantity = string.format("%d", count) }))
+					else
+						ui.text(et:GetName())
+					end
+				end
+			end
+		end
+	end
+end
+
+local function drawShipInfo()
+	local player = Game.player
+	local shipDef = ShipDef[player.shipId]
+	local hyperdrive = table.unpack(player:GetEquip("engine"))
+	local frontWeapon = table.unpack(player:GetEquip("laser_front"))
+	local rearWeapon = table.unpack(player:GetEquip("laser_rear"))
+
+	local mass_with_fuel = player.staticMass + player.fuelMassLeft
+	local mass_with_fuel_kg = 1000 * mass_with_fuel
+
+	-- ship stats mass is in tonnes; scale by 1000 to convert to kg
+	local fwd_acc = -shipDef.linearThrust.FORWARD / mass_with_fuel_kg
+	local bwd_acc = shipDef.linearThrust.REVERSE / mass_with_fuel_kg
+	local up_acc = shipDef.linearThrust.UP / mass_with_fuel_kg
+
+	-- delta-v calculation according to http://en.wikipedia.org/wiki/Tsiolkovsky_rocket_equation
```

Comment:
Sure I ruin your day and remove all the fun and joy from your life, but I feel there's already a (lot of) functions doing this calculation ;)

Reply:
My day remains unruined, and my life is still full of fun and joy. This is Web-eWorks' code :)

It probably makes sense to clear the first 2 views out and leave them empty like I did for the other ones.

Reply:
Actually, this isn't even _my_ code, but copied-and-pasted verbatim from the old `ShipInfo.lua` implementation. It was John Bartholomew's code, which itself was actually a refactor of dear old @nozmajner's code. If you need someone to blame, blame him! :stuck_out_tongue:

Reply:
Ahahah! ...Well, I don't wont to blame someone in particular because to me it seems, skimming trough code, everyone add its implementation of that equation :smile: ...

Just pinning that a beautiful thing could be to remove this and the others hundreds thousands methods to calculate deltaV and add a centralised Tsiolkovsky equation... Not mandatory right now: it may be one of the bigger refactoring to do ;)